pkg/cmd/roachtest/test_runner.go

```diff
@@ -328,19 +327,18 @@ func (r *testRunner) Run(
 		// Don't spin up more workers than necessary.
 		parallelism = n * count
 	}
-	for _, testSpec := range tests {
+	for i := range tests {
 		//  TODO(bhaskar): remove this once we have more usage details
 		//  and more convinced about using spot VMs for all the runs.
 		if roachtestflags.Cloud == spec.GCE &&
-			testSpec.Benchmark &&
-			!testSpec.Suites.Contains(registry.Weekly) &&
-			rand.Float64() <= 0.5 {
-			lopt.l.PrintfCtx(ctx, "using spot VMs to run test %s", testSpec.Name)
-			testSpec.Cluster.UseSpotVMs = true
+			tests[i].Benchmark &&
+			!tests[i].Suites.Contains(registry.Weekly) {
```

Comment:
Should we keep the 50% logic here? I think just fixing the cluster reuse case (which is what this is doing) is a good first step. We can make this unconditional in the future if we think it's the right move.